app/features/utils/functions.js

```diff
@@ -3,7 +3,33 @@
 // @flow
 
 import { shell } from 'electron';
-import config from '../config';
+
+/**
+ * Finds the external_api.js for the given Server URL. It also handles if the
+ * server URL contains tenant name.
+ *
+ * @param {string} serverURL - Jitsi Meet Server URL.
+ * @returns {Promise} - Promise which returns the external_api.js url.
+ */
+export async function findExternalApiURL(serverURL: string) {
+    const apiURL = getExternalApiURL(serverURL);
+
+    const response = await fetch(apiURL, { method: 'HEAD' });
+
+    if (response.ok) {
+        return apiURL;
+    }
+    // eslint-disable-next-line no-param-reassign
+    serverURL = serverURL.replace(/https?:\/\//, '');
```

Comment:
why do we need to do this? we are adding the scheme in normalizeServerURL (called from getExternalApiURL) and then removing it here?

Reply:
We need to strip it here because we need to calculate  `serverURL.lastIndexof('/')`.
If we don't it will find `/` in `https://`.

Reply:
can we use a URL object in fetch() ? If so, we could do that in url.pathname.

Reply:
> can we use a URL object in fetch() ? If so, we could do that in url.pathname.

I see what you are saying. It might look cleaner. I'll try that.

Reply:
> can we use a URL object in fetch() ? If so, we could do that in url.pathname.

@saghul 
Using the URL object is getting confusing and it does not reduce any amount of code. 
It is also forcing to create a new object and I actually want to avoid that.

Reply:
Hum, then let's try something else: process the URL and create all variations of it beforehand in an array, then process the items in turns, one at a time (from longest to shortest). This way we'd only fiddle with the URL parsing once, I think.

Reply:
You mean don't make it recursive?

IMHO, (and I am not being lazy 😄 ) it will just increase amount of code and complexity.

Reply:
I'd like us to not do and undo the HTTPS thing on the URL for each turn, please. If you have another option in mind, I'll be happy to hear it!

Reply:
@saghul 
We aren't undoing it. Is there a confusion?
The `getExternalApiURL` value goes to the `apiURL`.
And we are removing `https://` from `serverURL` and we only really need it to be in the first call. Not all recursive calls. But it checks whether to do it or not automatically.